addons/sourcemod/scripting/soap_tf2dm.sp

```diff
@@ -2082,3 +2098,32 @@ public OnPluginEnd()
 {
     MC_PrintToChatAll(SOAP_TAG ... "Soap DM unloaded.");
 }
+
+public bool GetConfigPath(const char[] map, char[] path, int maxlength)
+{
+    LogMessage("No config for: %s, searching for fallback", map);
+    char cleanMap[64];
+    strcopy(cleanMap, sizeof(cleanMap), map);
+    Regex normalizeMapRegex = new Regex("(_(a|b|beta|u|r|v|rc|f|final|comptf|ugc)?[0-9]*[a-z]?$)|([0-9]+[a-z]?$)", 0);
+    char match[64];
+
+    normalizeMapRegex.Match(cleanMap);
+    if (normalizeMapRegex.GetSubString(0, match, sizeof(match), 0)) {
+        ReplaceString(cleanMap, sizeof(cleanMap), match, "", true);
+    }
+    LogMessage("Cleaned map %s.", cleanMap);
+
+    BuildPath(Path_SM, path, maxlength, "configs/soap");
+    DirectoryListing dh = OpenDirectory(path);
+    char file[128];
+    bool foundMatch = false;
+    while (dh.GetNext(file, sizeof(file))) {
+        if (StrContains(file, cleanMap, false) == 0) {
```

Comment:
Did you mean `if (StrContains(file, cleanMap, false) != -1)`, because you'll exclude valid maps that don't match at char 0 otherwise

Reply:
This is intentional as the cleaned map will only match from the start anyway (it still starts with `cp_...`)